lib/ui/hash_codes.dart

```diff
@@ -105,19 +103,16 @@ int hashValues(
       }
     }
   }
-  return result;
+  return _Jenkins.finish(result);
 }
 
 /// Combine the hashCodes of an arbitrary number of values from an Iterable into
 /// one value. This function will return the same value if given "null" as if
 /// given an empty list.
 int hashList(Iterable<Object> args) {
-  int result = 373;
+  int result = 0;
   if (args != null) {
-    for (Object arg in args) {
-      assert(arg is! Iterable);
-      result = 37 * result + arg.hashCode;
-    }
+    for (Object arg in args) result = _Jenkins.combine(result, arg);
```

Comment:
nit: minor tweak here, please put a newline before the statement (between `)` and `result`).